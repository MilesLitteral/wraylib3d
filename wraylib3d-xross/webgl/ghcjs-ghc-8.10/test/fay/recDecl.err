recDecl.hs:14:6-18: Missing field in record construction i
