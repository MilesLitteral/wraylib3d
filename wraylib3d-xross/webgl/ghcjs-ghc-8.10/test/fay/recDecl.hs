data R = R { i :: Double, c :: Char } deriving Show
data S = S Double Char deriving Show

-- RecDecl
r1 :: R
r1 = R { i = 1, c = 'a' }

-- RecDecl with fields out of order
r2 :: R
r2 = R { c = 'b', i = 2 }

-- Partial RecDecl (Produces GHC warning)
r' :: R
r' = R { c = 'b' }

-- Regular application
r3 :: R
r3 = R 3 'c'

-- Using regular constructor
s1 :: S
s1 = S 1 'a'

-- Multiple fields with the same type
data X = X { _x1, _x2 :: Int } deriving Show

x1 = X 1 2
x2 = X { _x1 = 1, _x2 = 2 }

-- Record updates
r1' = r1{ i = 10 }
r2' = r2{ c = 'a', i = 20 }
r'' = r'{ i = 123 }

main = do
  -- print updated records first to show that old records are preserved
  print r1'
  print r2'
  print r''

  print r1
  putStrLn (show (i r1))
  putStrLn (show (c r1))
  print r2
  print r'
  print r3
  print s1
  print x1
  print x2
