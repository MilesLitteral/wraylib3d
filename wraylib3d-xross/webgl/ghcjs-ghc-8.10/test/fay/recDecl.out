R {i = 10.0, c = 'a'}
R {i = 20.0, c = 'a'}
R {i = 123.0, c = 'b'}
R {i = 1.0, c = 'a'}
1.0
'a'
R {i = 2.0, c = 'b'}
